### ISSUE TITLE
nvcc fatal   : A single input file is required for a non-link phase when an outputfile is specified

### ISSUE THREAD

**Issue by mreid2005** [KlausT/ccminer#263]
compiling 8.23 on Vs Studio 2015, with cuda  9.2
I've scoured, and scoured the compile settings, and I think I found what might be the issue, but I'm not sure if it is for sure, and 2, where to fix it 

It runs through the clean section then starts the compile, and I get 
the usual build cmdline string, and the error i referenced in the title.
when I started parsing the build cmdline I came across this.

Note the quotes inside the quotes for the -Xcompiler segment, without being escaped.
If this is causing the build to fail, I would like some help figuring out what I need to tweak to get past this point.

` -Xcompiler "/EHsc /W3 /nologo /Ox /Fd"C:\Users\mikey\Documents\Visual Studio 2015\Projects\ccminer-8.23\Release\temp\x64\vc140.pdb" /FS /Zi  /MT /GR-"`

**Complete Cmdline and ERROR -**

`C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\BuildCustomizations\CUDA 9.2.targets(712,9): error MSB3721: The command ""C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.2\bin\nvcc.exe" -gencode=arch=compute_60,code=\"sm_60,compute_60\" -gencode=arch=compute_61,code=\"sm_61,compute_61\" -gencode=arch=compute_52,code=\"sm_52,compute_52\" -gencode=arch=compute_50,code=\"sm_50,compute_50\" -gencode=arch=compute_35,code=\"sm_35,compute_35\" -gencode=arch=compute_30,code=\"sm_30,compute_30\" -gencode=arch=compute_37,code=\"sm_37,compute_37\" -gencode=arch=compute_70,code=\"sm_70,compute_70\" --use-local-env -ccbin "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\x86_amd64" -x cu  -I. -Icompat -Icompat\includes -I"compat\includes-x64" -Icompat\getopt -Icompat\includes\pthreads -I"compat\curl-for-windows\zlib" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.2\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.2\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.2\include"    --keep --keep-dir x64\Release\temp\x64 -maxrregcount=80  --machine 64 --compile -cudart shared -Wno-deprecated-gpu-targets    -DHAVE_STRUCT_TIMESPEC -DWIN32 -DNDEBUG -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES_COUNT -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES -D_CRT_SECURE_NO_WARNINGS -DCURL_STATICLIB -DUSE_WRAPNVML -DSCRYPT_KECCAK512 -DSCRYPT_CHACHA -DSCRYPT_CHOOSE_COMPILETIME -D_MBCS -Xcompiler "/EHsc /W3 /nologo /Ox /Fd"C:\Users\mikey\Documents\Visual Studio 2015\Projects\ccminer-8.23\Release\temp\x64\vc140.pdb" /FS /Zi  /MT /GR-" -o "C:\Users\mikey\Documents\Visual Studio 2015\Projects\ccminer-8.23\Release\temp\x64\x14.cpp.obj" "C:\Users\mikey\Documents\Visual Studio 2015\Projects\ccminer-8.23\x15\x14.cpp"" exited with code 1.`

`1>  nvcc fatal   : A single input file is required for a non-link phase when an outputfile is specified`

**Comment by KlausT** [comment]
On my system it looks like this:
`"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.2\bin\nvcc.exe" -ccbin "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\x86_amd64" -x cu  -I. -Icompat -Icompat\includes -I"compat\includes-x64" -Icompat\getopt -Icompat\includes\pthreads -I"compat\curl-for-windows\zlib" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.2\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.2\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.2\include"    --keep --keep-dir x64\Release\temp\x64 -maxrregcount=80  --machine 64 --compile  -Wno-deprecated-gpu-targets     -DHAVE_STRUCT_TIMESPEC -DWIN32 -DNDEBUG -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES_COUNT -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES -D_CRT_SECURE_NO_WARNINGS -DCURL_STATICLIB -DUSE_WRAPNVML -DSCRYPT_KECCAK512 -DSCRYPT_CHACHA -DSCRYPT_CHOOSE_COMPILETIME -D_MBCS -Xcompiler "/EHsc /W3 /nologo /Ox  /FS /Zi  /MT /GR-" -o D:\code\ccminer\Release\temp\x64\sia.cpp.obj "D:\code\ccminer\Sia\sia.cpp" -clean`

No /Fd option here for some reason.
The /Fd option is the "Program Database Filename".
It's set to $(IntDir)vc$(PlatformToolsetVersion).pdb here (the default)

I wonder if the compiler doesn't use quotes if the path doesn't contain spaces.

**Comment by KlausT** [comment]
Oh, sorry, wrong one.
This is the command line here:

`"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin\nvcc.exe" -gencode=arch=compute_60,code=\"sm_60,compute_60\" -gencode=arch=compute_61,code=\"sm_61,compute_61\" -gencode=arch=compute_52,code=\"sm_52,compute_52\" -gencode=arch=compute_50,code=\"sm_50,compute_50\" -gencode=arch=compute_35,code=\"sm_35,compute_35\" -gencode=arch=compute_30,code=\"sm_30,compute_30\" -gencode=arch=compute_37,code=\"sm_37,compute_37\" -gencode=arch=compute_70,code=\"sm_70,compute_70\" -gencode=arch=compute_75,code=\"sm_75,compute_75\" --use-local-env -ccbin "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\x86_amd64" -x cu  -I. -Icompat -Icompat\includes -I"compat\includes-x64" -Icompat\getopt -Icompat\includes\pthreads -I"compat\curl-for-windows\zlib" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\include"    --keep --keep-dir x64\Release\temp\x64 -maxrregcount=80  --machine 64 --compile -cudart shared -Wno-deprecated-gpu-targets    -DHAVE_STRUCT_TIMESPEC -DWIN32 -DNDEBUG -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES_COUNT -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES -D_CRT_SECURE_NO_WARNINGS -DCURL_STATICLIB -DUSE_WRAPNVML -DSCRYPT_KECCAK512 -DSCRYPT_CHACHA -DSCRYPT_CHOOSE_COMPILETIME -D_MBCS -Xcompiler "/EHsc /W3 /nologo /Ox /FdD:\code\ccminer\Release\temp\x64\vc140.pdb /FS /Zi  /MT /GR-" -o D:\code\ccminer\Release\temp\x64\x14.cpp.obj "D:\code\ccminer\x15\x14.cpp"`

Looks like having spaces in the path is not a good idea.

**Comment by mreid2005** [comment]
Oh for heaven's sake! I have tried to compile so many iterations of ccminer over the last year, just for the sake of learning, and it wasn't until this last two weeks. I decided I wasn't going to quit until I figured it out. So for the last two weeks, I've spent at least an hour or two every other night, pouring over log files, and doing my best to not open an "issue" over something that seemed like it was so straightforward for everyone else.

So.. A big hearty Thank you! because sure 'nuff that was the problem!!!!!
